src/shared/mkfs-util.c

```diff
@@ -388,6 +388,11 @@ int make_filesystem(
                 if (quiet && strv_extend(&argv, "-q") < 0)
                         return log_oom();
 
+                /* mkfs.btrfs unconditionally warns about several settings changing from v5.15 onwards which
+                 * isn't silenced by "-q", so let's redirect stdout to /dev/null as well. */
+                if (quiet)
+                        stdio_fds[1] = -EBADF;
```

Comment:
hmm, wouldn't it be nicer to just add arguments to our invocation that match the defaults on v5.15 onwards? that we we get same behaviour everywhere, and no warnings.

Reply:
what? that's stupid... they warn about issues the mkfs invocation is not actually suffering by? hmpf

Reply:
hmm, i wonder if we really should do this though, isn't this taping over some fuckup in mkfs.btrfs?

at the very least they should make -q silence this

Reply:
if they fix it we can revise later, this is annoying to deal with otherwise